src/lib/server/services/content.ts

```diff
@@ -1,8 +1,185 @@
-import { and, count, countDistinct, eq, gte, sql } from 'drizzle-orm';
+import { count, countDistinct, eq, sql, desc, asc, isNull, not, inArray, and, gte } from 'drizzle-orm';
 import type { DB } from '..';
 import { postTable } from '../db/posts.sql';
 import { channelTable } from '../db/channels.sql';
 import { postVoteTable } from '../db/votes.posts.sql';
+import type { PostData, uuid } from '$lib/types';
+import { userTable } from '../db/users.sql';
+import { postTagTable } from '../db/tags.posts.sql';
+import { channelTagsTable } from '../db/tags.channels.sql';
+import { commentTable } from '../db/comments.sql';
+import { publicChannelTable } from '../db/public.channels.sql';
+import { PAGE_SIZE } from '$lib';
+import { array_agg, dedupe_nonull_array } from './utils';
+import { userBlockTable } from '../db/blocks.users.sql';
+import { subscriptionTable } from '../db/subscriptions.sql';
+
+interface GenericPostFilter {
+    requesterId?: uuid;
+    sort: 'votes' | 'date';
+    filter: 'subscribed' | 'all';
+    reverseSort?: boolean;
```

Comment:
likely want date controls.

It would be a questionable user experience if the pages of data were changing as I'm scrolling and I end up seeing duplicate posts. So it makes sense to only show posts with timestamps <= the timestamp of when they initially hit the page.